src/SwedbankPay/Api/Client/ClientVersion.php

```diff
@@ -229,35 +229,52 @@ private function tryReadComposerJson(&$decodedJsonObject) : bool
      * composer package root path because we might be part of another composer install.
      * Returns true if successful; otherwise false.
      *
-     * @param object $decodedJsonObject The by-reference $decodedJsonObject variable to assign the JSON decoded object to.
+     * @param object $decodedJsonObject The by-reference variable to assign the JSON decoded object to.
      * @param string $path Optional alternative path to look for composer lock file.
      * @return bool true if the JSON decoding is successful; otherwise false.
      */
     private function tryReadComposerLock(&$decodedJsonObject, $path = '') : bool
     {
-        if ($path == '') {
-            $path = $this->getComposerPath() . DIRECTORY_SEPARATOR . 'composer.lock';
+        if (!is_readable($path)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function is_readable() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5631375)